lib/Runtime/Debug/DiagObjectModel.cpp

```diff
@@ -3837,4 +3880,195 @@ namespace Js
         return FALSE;
     }
 #endif
+
+    //--------------------------
+    // RecyclableSimdObjectWalker
+
+    template <typename simdType, uint elementCount>
+    BOOL RecyclableSimdObjectWalker<simdType, elementCount>::Get(int i, ResolvedObject* pResolvedObject)
+    {
+        Assert(elementCount == 4 || elementCount == 8 || elementCount == 16); // SIMD types such as int32x4, int8x16, int16x8
+        Assert(i >= 0 && i <= elementCount);
+
+        simdType* simd = simdType::FromVar(instance);
+        SIMDValue value = simd->GetValue();
+
+        WCHAR* indexName = AnewArray(GetArenaFromContext(scriptContext), WCHAR, SIMD_INDEX_VALUE_MAX_NAME);
+        swprintf_s(indexName, SIMD_INDEX_VALUE_MAX_NAME, L"[%d]", i);
+        pResolvedObject->name = indexName;
+
+        TypeId simdTypeId = JavascriptOperators::GetTypeId(instance);
+
+        switch (simdTypeId)
+        {
+        case TypeIds_SIMDInt32x4:
+            pResolvedObject->obj = JavascriptNumber::ToVarWithCheck(value.i32[i], scriptContext);
+            break;
+        case TypeIds_SIMDFloat32x4:
+            pResolvedObject->obj = JavascriptNumber::ToVarWithCheck(value.f32[i], scriptContext);
+            break;
+        case TypeIds_SIMDInt8x16:
+            pResolvedObject->obj = JavascriptNumber::ToVarWithCheck(value.i8[i], scriptContext);
+            break;
+        case TypeIds_SIMDInt16x8:
+            pResolvedObject->obj = JavascriptNumber::ToVarWithCheck(value.i16[i], scriptContext);
+            break;
+        case TypeIds_SIMDBool32x4:
+            pResolvedObject->obj = JavascriptBoolean::ToVar(value.i32[i], scriptContext);
+            break;
+        case TypeIds_SIMDBool8x16:
+            pResolvedObject->obj = JavascriptBoolean::ToVar(value.i8[i], scriptContext);
+            break;
+        case TypeIds_SIMDBool16x8:
+            pResolvedObject->obj = JavascriptBoolean::ToVar(value.i16[i], scriptContext);
+            break;
+        case TypeIds_SIMDUint32x4:
+            pResolvedObject->obj = JavascriptNumber::ToVarWithCheck(value.u32[i], scriptContext);
+            break;
+        case TypeIds_SIMDUint8x16:
+            pResolvedObject->obj = JavascriptNumber::ToVarWithCheck(value.u8[i], scriptContext);
+            break;
+        case TypeIds_SIMDUint16x8:
+            pResolvedObject->obj = JavascriptNumber::ToVarWithCheck(value.u16[i], scriptContext);
+            break;
+        default:
+            AssertMsg(false, "Unexpected SIMD typeId");
+            return FALSE;
+        }
+
+        pResolvedObject->propId = Constants::NoProperty;
+        pResolvedObject->scriptContext = scriptContext;
+        pResolvedObject->typeId = simdTypeId;
+        pResolvedObject->objectDisplay = pResolvedObject->CreateDisplay();
+        pResolvedObject->objectDisplay->SetDefaultTypeAttribute(DBGPROP_ATTRIB_VALUE_READONLY | DBGPROP_ATTRIB_VALUE_IS_FAKE);
```

Comment:
Yes, I think it's by design at this stage, we don't want user to enter invalid index to cause problem, just want to display the correct SIMD value for each index for now.